bin/libs/component.sh

```diff
@@ -408,6 +408,154 @@ process_zss_plugin_install() {
+    zwes_zis_pluginlib=$(read_yaml "${ZWE_CLI_PARAMETER_CONFIG}" ".zowe.setup.dataset.authPluginLib")
+    zwes_zis_parmlib=$(read_yaml "${ZWE_CLI_PARAMETER_CONFIG}" ".zowe.setup.dataset.parmlib")
+    zwes_zis_parmlib_member="ZWESIP00"
+    print_trace "- Checking for zis plugins and verifying them"
+    component_dir="${1}"
+    
+    iterator_index=0
+    zis_plugin_id=$(read_component_manifest "${component_dir}" ".zisPlugins[${iterator_index}].id" 2>/dev/null)
+    zis_plugin_path=$(read_component_manifest "${component_dir}" ".zisPlugins[${iterator_index}].path" 2>/dev/null)
+    while [ -n "${zis_plugin_path}" ]; do
+      cd "${component_dir}"
+      zis_plugin_install "${zis_plugin_path}" "${zwes_zis_pluginlib}" "${zwes_zis_parmlib}" "${zwes_zis_parmlib_member}" "${zis_plugin_id}" "${component_dir}"
+
+      iterator_index=`expr $iterator_index + 1`
+      zis_plugin_path=$(read_component_manifest "${component_dir}" ".zisPlugins[${iterator_index}].id" 2>/dev/null)
+      zis_plugin_id=$(read_component_manifest "${component_dir}" ".zisPlugins[${iterator_index}].id" 2>/dev/null)
+    done
+  fi
+}
+
+# $1 - needle
+# $2 - haystack
+is_substr_of() {
+  is_substring=$(echo "$2" | grep -c "$1")
+  return "$is_substring"
+}
+
+# $1 - file_name
+# $2 - data_set_name
+copy_uss_to_mvs() {
+  cp "$1" "//'$2'"
+  return $?
+}
+
+# $1 = data_set_name
+# $2 = file_name
+copy_mvs_to_uss() {
+  cp "//'$1'" "$2"
+  return $?
+}
+
+# $1 = key=value
+get_key_of_string() {
+  echo "$1" | sed 's/=/ /g' | awk '{print $1}'
+}
+
+# $1 = key=value
+get_value_of_string() {
+  echo "$1" | sed 's/=/ /g' | awk '{print $2}'
+}
+
+# $1 = line number
+# $2 = file
+get_string_at_line_number() {
+  awk -v n="$1" 'NR == n {print $0}' "$2"
+}
+
+# $1 = search_key
+# $2 = file
+get_line_number_of_key() {
+  grep "$1" "$2" > /dev/null
+  if [ $? -eq 0 ]; then
+    awk -v s="$1" '$0 ~ s {print NR}' "$2"
+  else
+    echo ""
+  fi
+}
+
+# $1 = line number
+# $2 = file
+remove_key_value_at_line_number() {
+  mv "$2" "$2.tmp"
+  awk -v n=$1 'NR != n' "$2.tmp" > "$2"
+  rm -f "$2.tmp"
+}
+
+# $1 = key=value
+# $2 = file
+add_key_value_at_end_of_file() {
+  echo "$1" >> "$2"
+}
+
+zis_plugin_install() {
+  plugin_path="${1}"
+  zwes_zis_pluginlib="${2}"
+  zwes_zis_parmlib="${3}"
+  zwes_zis_parmlib_member="${4}"
+  plugin_id="${5}"
+  component_dir="${6}"
+  parmlib_member_as_unix_file="./${zwes_zis_parmlib_member}.txt"
+
+  if [ ! -f "$parmlib_member_as_unix_file" ]; then
+    copy_mvs_to_uss "${zwes_zis_parmlib}(${zwes_zis_parmlib_member})" "${parmlib_member_as_unix_file}"
+  fi
+
+  changed=0
+
+  if [ -d "${component_dir}${plugin_path}" ]; then
+    if [ -d "${component_dir}${plugin_path}/loadlib" ] && [ -d "${component_dir}${plugin_path}/samplib" ]; then
+      for module in $(ls ${component_dir}${plugin_path}/loadlib); do
+        copy_uss_to_mvs "${component_dir}${plugin_path}/loadlib/${module}" "$zwes_zis_pluginlib"
+        if [ $? != 0 ]; then
+          exit 1
```

Comment:
We shouldn't `exit` but should write error message and `return`, then let caller to decide what to do.

Reply:
A unique exit code is recommended and should documented in `.errors` file. For example,

```
print_error "Error ZWEL0200E: failed to copy USS file %s to MVS data set %s."
return 200
```